### PR TITLE
MINOR: Add support for bootstrap-server in create_topic

### PR DESCRIPTION
**Pull request by soondenana**
create_topic method uses --zookeeper option to create a new topic. This
change adds support for it to switch to --bootstrap-server config if it
is supported.

As "--zookeeper" option is deprecated and "--bootstrap-server" is
preferred, this change allows us to move away from the deprecated
option.

Tested by launching tests and confirming they pass and use the new
--bootstrap-server option.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by soondenana** [comment]
cc @mumrah , @jsancio, @hachikuji

File: tests/kafkatest/services/kafka/kafka.py [hunk 1]
```diff
@@ -746,6 +763,18 @@ def describe_consumer_group(self, group, node=None, command_config=None):
...
 
+    def connect_setting(self, node):
+        """
+        Checks if --bootstrap-server config is supported, if yes then returns a string with
+        bootstrap server, otherwise returns zookeeper connection string.
+        """
+        if node.version.supports_bootstrap_server():
+            connection_setting = "--bootstrap-server %s" % self.bootstrap_servers(self.security_protocol)
+        else:
+            connection_setting = "--zookeeper %s" % self.zk_connect_setting()
```
Comment (jsancio) [inline tests/kafkatest/services/kafka/kafka.py]: Or we can support `--bootstrap-server` only. Wouldn't `node.version.supports_bootstrap_server()` always return true?

File: tests/kafkatest/services/kafka/kafka.py [hunk 2]
```diff
@@ -487,6 +489,20 @@ def set_unclean_leader_election(self, topic, value=True, node=None):
         self.logger.info("Running alter unclean leader command...\n%s" % cmd)
         node.account.ssh(cmd)
 
+    def perform_leader_election(self, topic, partition, election_type, node=None):
+        """
+        Perform leader election for the partition of the topic passed in as argument.
```
Comment (jsancio) [inline tests/kafkatest/services/kafka/kafka.py]: Maybe this should read:
> Perform leader election for the partition of the topic passed in as argument and waits until leadership changes replica.

**Review by jsancio — APPROVED** [review]
LGTM. Left some minor comments.

File: tests/kafkatest/services/kafka/kafka.py (hunk 1 above)
Comment (soondenana) [reply tests/kafkatest/services/kafka/kafka.py]: @hachikuji  mentioned that these tests can run against a lower version of Kafka cluster, so it is possible for it to run against say 2.1 version.

File: tests/kafkatest/services/kafka/kafka.py (hunk 2 above)
Comment (soondenana) [reply tests/kafkatest/services/kafka/kafka.py]: Done.

File: tests/kafkatest/services/kafka/kafka.py (hunk 1 above)
Comment (jsancio) [reply tests/kafkatest/services/kafka/kafka.py]: Hmmm. If that is the case then at least `kafka-leader-election.sh` won't exist in older versions of Kafka. I think we added it in 2.3.

Comment (soondenana) [reply tests/kafkatest/services/kafka/kafka.py]: Talked to @hachikuji  and it seems that the test in discussion is client_compatibility_features_test.py. I looked at the test and I think you are right, we don't need older zookeeper connection. I will update the code as you suggested and post a new commit.

**Comment by soondenana** [comment]
Closing this as code has diverged and some of it is already been taken care of as part of KIP-500.

**Pull request closed without merging**